hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -195,6 +195,118 @@ data Environment = Environment
   , otherParties :: [Party]
   }
 
+--
+data L2NextState tx
+  = SameState
+  | NextInitialState {pendingCommits :: PendingCommits, committed :: Committed tx}
+  | NextOpenState {coordinatedHeadState :: CoordinatedHeadState tx}
+  | NextClosedState {confirmedSnapshot :: ConfirmedSnapshot tx}
```

Comment:
If it's about switching between Initial, Open and close I would not call this an `L2` state. The `CoordinatedHeadState` is basically what we could call the `L2State` as it is only about the status of the off-chain/L2 protocol.